core/src/test/scala/org/broadinstitute/dsde/rawls/googleProject/GoogleProjectRegistrationServiceSpec.scala

```diff
@@ -526,4 +507,71 @@ class GoogleProjectRegistrationServiceSpec
 
   }
 
+  it should "throw an error if updating the billing account fails" in {
+    val mockSamDAO = mock[SamDAO]
+    val mockGoogleProjectRegRepo = mock[GoogleProjectRegistrationRepository]
+    val mockContext = mock[RawlsRequestContext]
+    val mockBillingRepository = mock[BillingRepository]
+    val mockGoogleServicesDAO = mock[GoogleServicesDAO]
+
+    when(mockGoogleServicesDAO.setBillingAccountName(any[GoogleProjectId], any[RawlsBillingAccountName], any()))
+      .thenThrow(new RuntimeException("Something has gone wrong in Google"))
+
+    when(mockGoogleProjectRegRepo.getGoogleProjectRegistration(any[GoogleProjectId]))
+      .thenReturn(Future.successful(None))
+
+    val billingProjectId = RawlsBillingProjectName("billing-project-id")
+
+    val billingProject: RawlsBillingProject = RawlsBillingProject(UUID.randomUUID(),
+                                                                  billingProjectId,
+                                                                  CreationStatuses.Ready,
+                                                                  Option(RawlsBillingAccountName("billing-account")),
+                                                                  None
+    )
+
+    when(mockBillingRepository.getBillingProject(mockitoEq(billingProjectId)))
+      .thenReturn(Future.successful(Some(billingProject)))
+
+    when(
+      mockSamDAO.userHasAction(any[SamResourceTypeName], any[String], any[SamResourceAction], any[RawlsRequestContext])
+    ).thenReturn(Future.successful(true))
+
+    when(
+      mockSamDAO.listUserActionsForResource(mockitoEq(SamResourceTypeNames.billingProject),
+                                            any[String],
+                                            any[RawlsRequestContext]
+      )
+    ).thenReturn(
+      Future.successful(Set(SamBillingProjectActions.link, SamBillingProjectActions.own))
+    )
+
+    val googleProjectRegService =
+      GoogleProjectRegistrationService.constructor(
+        mockSamDAO,
+        mockGoogleProjectRegRepo,
+        mockBillingRepository,
+        mockGoogleServicesDAO
+      )(
+        mockContext
+      )
+
+    val testProject = GoogleProjectRegistration(GoogleProjectId("test-project"),
+                                                Some(RawlsBillingAccountName("billing-account")),
+                                                None,
+                                                billingProjectId
+    )
+    when(mockGoogleProjectRegRepo.registerGoogleProject(any[GoogleProjectRegistration]))
+      .thenReturn(Future.successful(testProject))
+
+    val e = intercept[RuntimeException] {
```

Comment:
This will work, but I'd prefer to be more specific with the exception that we're expecting and only intercept `RawlsExceptionWithErrorReport`